source/MaterialXTest/Runtime.cpp

```diff
@@ -417,15 +417,17 @@ TEST_CASE("Runtime: CoreIo", "[runtime]")
 
         // Write the full stage to a new document
         // and save it to file for inspection.
-        stageIo.write(stage.getName().str() + "_export.mtlx", nullptr, nullptr);
+        stageIo.write(stage.getName().str() + "_export.mtlx", nullptr);
 
         // Write only nodegraphs to a new document
         // and save it to file for inspection.
-        auto nodeGraphFilter = [](const mx::RtObject& obj) -> bool
+        mx::RtWriteOptions writeOptions;
+        writeOptions.writeFilter = [](const mx::RtObject& obj) -> bool
         {
             return obj.hasApi(mx::RtApiType::NODEGRAPH);
```

Comment:
@JGamache-autodesk : for your requirements to export a node graph, this will allow extraction of all node-graphs but no a specific one. We can figure out what is best here. e.g. Allow for an option to specify a "root" node to start from.